doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,13 @@
 
 <h3>New features since last release</h3>
 
+* Added a modified version of the `_simplify` function to the `hf` module.
+  [(#2103)](https://github.com/PennyLaneAI/pennylane/pull/2103)
+
+  This functions combines redundant terms in a Hamiltonian and eliminates terms with a zero
+  coefficient. The new function makes construction of molecular Hamiltonians more efficient.
+  For LiH, as an example, the time to construct the Hamiltonian is reduced from ~200 s to ~10 s.
```

Comment:
```suggestion
  For LiH, as an example, the time to construct the Hamiltonian is reduced roughly by a factor of 20.
```
Since runtimes depend on the device running the code